contrib/mod_deflate.c

```diff
@@ -513,30 +508,6 @@ static int deflate_netio_read_cb(pr_netio_stream_t *nstrm, char *buf,
     switch (deflate_zerrno) {
       case Z_OK:
       case Z_STREAM_END:
-        copylen = deflate_zbufsz - zstrm->avail_out;
```

Comment:
With proper handling of the input and output buffers, we no longer need to handle the edge case of dynamically detecting (and wrongly in this case) of when we need to increase the size of the output buffer.